### PR TITLE
Update vobject to 3.4.2

### PR DESCRIPTION
**Pull request by libasys**
With this change  the vobject is compatibel with Version 3.4.2

### CODE REVIEW

File: lib/private/vobject.php [hunk 1]
```diff
@@ -101,7 +101,25 @@ public function __construct($vobject_or_name) {
...
+				case 'VFREEBUSY':	
+				case 'VJOURNAL':
+				case 'VTIMEZONE':					
+					$this->vcomponent = new Sabre\VObject\Component\VCalendar();
+					break;	
+				case 'VCARD':
+					$this->vcomponent = new Sabre\VObject\Component\VCard();
+					break;
+				default:
+				 	$this->vcomponent = new Sabre\VObject\Component\VCalendar();
```
Comment (nickvergessen) [inline lib/private/vobject.php]: Move the `case 'VCARD':` block to the top, then you don't need to list the remaining cases, or at least you can put them in one block

File: lib/private/vobject.php [hunk 2]
```diff
@@ -163,20 +185,28 @@ public function setString($name, $string) {
 	 *
 	 * @param string $name
 	 * @param DateTime $datetime
-	 * @param int $dateType
+	 * @param  $floating
```
Comment (nickvergessen) [inline lib/private/vobject.php]: missing type

File: lib/private/vobject.php [hunk 3]
```diff
@@ -163,20 +185,28 @@ public function setString($name, $string) {
...
 		if ($datetime == 'now') {
-			$datetime = new DateTime();
+			$datetime = new \DateTime();
 		}
-		if ($datetime instanceof DateTime) {
-			$datetime_element = new Sabre\VObject\Property\DateTime($name);
-			$datetime_element->setDateTime($datetime, $dateType);
-			$this->vObject->__set($name, $datetime_element);
+		if ($datetime instanceof \DateTime) {
+				$tThis=$this;
```
Comment (nickvergessen) [inline lib/private/vobject.php]: intendation is off

File: lib/private/vobject.php [hunk 4]
```diff
@@ -31,6 +31,7 @@
 class OC_VObject{
```
Comment (DeepDiver1975) [inline lib/private/vobject.php]: is that class needed somewhere in core? AFAIK this is calendar and contacts only ... time to move????

**Comment by libasys** [comment]
We can remove this class from core, cause the current calendar app doesn't works with OC 8.1 and vobject 3.4.xx! So the devs of calendar, contacts and task app have to write their own vobject class in their apps namespace!

**Comment by PVince81** [comment]
AFAIK the new VObject library isn't compatible any more with the way how core's VObject tries to override stuff. We probably want to move away from that core class.

**Comment by PVince81** [comment]
Here is the PR to remove that class: https://github.com/owncloud/core/pull/15691

@libasys the calendar app has been fixed yesterday to work again with master.

To be honest, I wasn't aware that it would have been possible to fix calendar/contacts/tasks by just changing this helper class. But in general I'm more in favor of not having this helper class in the first place. Also, SabreDAV 2.1 now should already contain all the fixes that were introduced through this class. (also this class is private, so it shouldn't even be used outside of core!)

**Comment by DeepDiver1975** [comment]
> Here is the PR to remove that class: #15691

and therefore closing this PR - THX @libasys for supplying the patch - nevertheless

**Pull request closed without merging**